File: 2021/src/test/java/nel/marco/Day4Test.java
package nel.marco;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.List;
import java.util.Map;

import static org.junit.Assert.assertEquals;

class Day4Test {

    Day4 day4 = new Day4();


    @Test
    void getAllBingoBoards() throws IOException {
        Map<Integer, int[][]> allBingoBoards = day4.getAllBingoBoards();


        assertEquals(99, allBingoBoards.size());
    }


    @Test
    void getRandomNumbers() throws IOException {
        List<Integer> randomNumbers = day4.getRandomNumbers();


        assertEquals(100, randomNumbers.size());
    }

    @Test
    void part1() throws IOException {

        Map<Integer, int[][]> allBingoBoards = day4.getAllBingoBoards();
        List<Integer> randomNumbers = day4.getRandomNumbers();
        long answer = day4.part1(randomNumbers, allBingoBoards);
        assertEquals(51776,answer);
    }


    @Test
    void part2() throws IOException {

        Map<Integer, int[][]> allBingoBoards = day4.getAllBingoBoards();
        List<Integer> randomNumbers = day4.getRandomNumbers();
        long answer = day4.part2(randomNumbers, allBingoBoards);
        assertEquals(16830,answer);
    }
}

File: 2024/src/test/nel/marco/Day4Test.kt
package nel.marco

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotEquals
import org.junit.jupiter.api.RepeatedTest
import org.junit.jupiter.api.Test

class Day4Test {


    val day = Day4(useExample = false, useMac = false)
    val example = Day4(useExample = true, useMac = false)


    @Test
    fun answerOne() {
        assertEquals("18", example.answerOne())
        assertEquals("2578", day.answerOne())
    }


    @Test
    fun answerTwo() {
        assertEquals("9", example.answerTwo())
        assertEquals("1972", day.answerTwo())
    }


    fun avg() {
        val day = day
        MarcoUtil.avgTime("part1",5) {
            assertEquals("2578", day.answerOne())
        }
        MarcoUtil.avgTime("part2",5) {
            assertEquals("1972", day.answerTwo())
        }
    }
}